### PR TITLE
HTTPS Support

### PR DESCRIPTION
**Pull request by bryaakov**
Related to #938 
- Still not implemented clusterDefault

### CODE REVIEW

File: server/cmwell-bg/src/main/scala/cmwell/bg/Merger.scala [hunk 1]
```diff
@@ -127,13 +127,13 @@ class Merger(config: Config) extends LazyLogging {
                            fields: Map[String, Set[FieldValue]],
                            lastModified: DateTime): Infoton = {
     prev_infoton match {
```
Comment (bryaakov) [inline server/cmwell-bg/src/main/scala/cmwell/bg/Merger.scala]: All of these need to preserve prev_infoton's protocol

File: server/cmwell-common/src/main/scala/cmwell/common/formats/JsonSerializerForES.scala [hunk 2]
```diff
@@ -155,13 +155,15 @@ object JsonSerializerForES extends AbstractJsonSerializer with NsSplitter with L
     if (current) jsonGenerator.writeBooleanField("current", true)
     else jsonGenerator.writeBooleanField("current", false)
 
+    infoton.protocol.foreach(jsonGenerator.writeStringField("protocol", _))
```
Comment (e-orz) [inline server/cmwell-common/src/main/scala/cmwell/common/formats/JsonSerializerForES.scala]: always provide the protocol not just for https

File: server/cmwell-common/src/test/scala/cmwell/common/InfotonSerializationSpecification.scala [hunk 3]
```diff
@@ -73,7 +73,7 @@ object InfotonSerializationSpecification extends Properties("Infoton") {
           case false => None
           case true => Some(mkFields)
         }
-        FileInfoton(path=path.mkString("/", "/", ""),dc="dc_test", fields=f, content=Some(FileContent(content, mimeType)))
+        FileInfoton(path=path.mkString("/", "/", ""),dc="dc_test", fields=f, content=Some(FileContent(content, mimeType)), protocol = None)
```
Comment (e-orz) [inline server/cmwell-common/src/test/scala/cmwell/common/InfotonSerializationSpecification.scala]: add test for https

File: server/cmwell-common/src/test/scala/cmwell/common/CommandSpec.scala [hunk 4]
```diff
@@ -109,7 +110,7 @@ class CommandSpec extends FlatSpec with Matchers with LazyLogging {
 
   "file infoton decode" should "be successful" in {
     System.setProperty("dataCenter.id", "dc_test")
-    val fInfoton = FileInfoton(path = "/stam/kacha", dc = "dc_test", content = Some(FileContent("test text".getBytes("UTF-8"), "text/plain")))
+    val fInfoton = FileInfoton(path = "/stam/kacha", dc = "dc_test", content = Some(FileContent("test text".getBytes("UTF-8"), "text/plain")), protocol = None)
```
Comment (e-orz) [inline server/cmwell-common/src/test/scala/cmwell/common/CommandSpec.scala]: add test for https

File: server/cmwell-ws/app/ld/util/LDFormatParser.scala [hunk 5]
```diff
@@ -822,6 +826,10 @@ object LDFormatParser extends LazyLogging {
             feedbacks += s"skipping validation for all ingested subjects"
           }
 
+//          cmwMetaDataMap.foreach { case (subject, md) =>
```
Comment (e-orz) [inline server/cmwell-ws/app/ld/util/LDFormatParser.scala]: remove commented out code

**Pull request merged**